### PR TITLE
Hegelc cleanup

### PR DESCRIPTION
**Pull request by displague**
## Description

* Moves `/hegelc` to `/cmd/hegelc`
* Adds environment variables (`HEGEL_SERVER`, `HEGEL_PORT`)
* Adds flags (`-server`, `-port`)
* changes default hostname to `metadata` from `metadata.packet.net`
* default error behavior is `Fatal` (exit code 1) rather than `Panic` (exit code 2)
* use `grpc.DialContext` instead of `grpc.Dial`
* fixes gocritic warnings about [exitAfterDefer](https://go-critic.com/overview.html#exitafterdefer)

## Why is this needed

Fixes: #67 

## How Has This Been Tested?

ran with `-help`, env variables set, flags set. Attempted invalid ports and hostnames and got appropriate errors.

## How are existing users impacted? What migration steps/scripts do we need?

The default server changes from metadata.packet.net to metadata.

## Checklist:

I have:

- [x] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

**Comment by mmlb** [comment]
Note that log.Panic panics and thus defered functions are run, but log.Fatal I think calls os.Exit and does not run defers. I think that's fine though and better for a cli than dumping a stack trace anyway.

File: cmd/hegelc/main.go [hunk 1]
```diff
@@ -4,36 +4,65 @@ import (
...
+)
+
 func main() {
+	if envServer := os.Getenv(envVarServer); envServer != "" {
+		defaultServer = envServer
+	}
+	if envPort := os.Getenv(envVarPort); envPort != "" {
+		var err error
+		if defaultPort, err = strconv.Atoi(envPort); err != nil {
+			log.Panic(err)
```
Comment (mmlb) [inline cmd/hegelc/main.go:37]: this should be log.Fatal too right?

Comment (mmlb) [inline cmd/hegelc/main.go:37]: fyi: github.com/packethost/pkg/env has a bunch of nice funcs that do these conversions, you might want to use that, but not going to require it.

**Review by mmlb — CHANGES_REQUESTED** [review]

File: cmd/hegelc/main.go (hunk 1 above)
Comment (displague) [reply cmd/hegelc/main.go]: The linter caught me and I changed more of this around.

**Review by mmlb — APPROVED** [review]
not a fan of basically empty main, but oh well not really a big deal either

**Pull request merged**